csi/csi.go

```diff
@@ -0,0 +1,76 @@
+/*
+CSI Interface for OSD
+Copyright 2017 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package csi
+
+import (
+	"net"
+	"sync"
+
+	//"github.com/libopenstorage/openstorage/api/spec"
+	csi "github.com/container-storage-interface/spec/lib/go/csi"
+
+	"go.pedge.io/dlog"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+type OsdCsiServerConfig struct {
+	Listener net.Listener
```

Comment:
If this is a config object shouldn't it just have ip and port number instead of a Listener object? This server can then create a listener object.

Reply:
By providing "Listener" the caller can create either a TCP or Unix Domain socket connection.

Reply:
Then maybe also have helper function like NewOcdCsiTcpServer() and NewOsdCsiSocketServer() which will create the listeners with sane defaults?

Reply:
agree with the comments above - passing Listener seems a error prone as well.